File: old-files/object-loop.py
json_object = json.loads('{"first_key":"first_value", "second_key":"second_value"}')
pairs = json_object.items()

for key, value in pairs:
    print(value)


for i in range(4):
    print("i =", i)


for(var i=0; i<4; i++) {
    console.log("i =", i);
}



# https://stackoverflow.com/questions/14696261/declaring-two-variables-in-a-for-loop
var i,c;
for(i = 0,c=aString.charAt(0); i < aString.length; ++i, c = aString.charAt(i)){
    alert("c: "+c)
    func1[typeOfChar(c)]++
}

# https://stackoverflow.com/questions/7241878/for-in-loops-in-javascript-key-value-pairs
#  https://attacomsian.com/blog/javascript-iterate-objects#:~:text=keys()%20Method-,The%20Object.,the%20value%20of%20each%20property.
for (var k in target){
    if (typeof target[k] !== 'function') {
         alert("Key is " + k + ", value is" + target[k]);
    }
}

File: archived-files/loop-and-test-archive/object-loop.js
// json_object = json.loads('{"first_key":"first_value", "second_key":"second_value"}')
// pairs = json_object.items()

// for key, value in pairs:
//     print(value)

//  Question:
let students = [
    {
        name: 'John',
        subjects: ['maths', 'english', 'cad'],
        teacher: {maths: 'Harry', english: 'Joan', cad: 'Paul'},
        results: {maths: 90, english: 75, cad: 87},
    },
    {
        name: 'Emily',
        subjects: ['science', 'maths','english', 'art'],
        teacher: {science: 'Iris', maths: 'Harry', english: 'Joan', art: 'Simon'},
        results: {science: 93, maths: 95, english: 80, art: 95},
    },
    {
        name: 'Adam',
        subjects: ['science', 'maths', 'art'],
        teacher: {science: 'Iris', maths: 'Harry', art: 'Simon'},
        results: {science: 63, maths: 79, art: 95},
    }
];

// Answer:


// https://stackoverflow.com/questions/7241878/for-in-loops-in-javascript-key-value-pairs
//  https://attacomsian.com/blog/javascript-iterate-objects#:~:text=keys()%20Method-,The%20Object.,the%20value%20of%20each%20property.
for (var k in students){
    if (typeof students[k] !== 'function') {
            alert("Key is " + k + ", value is" + students[k]);
    }
}
